UI/States/CraftingUIState.cs

```diff
@@ -1206,30 +1206,57 @@ or ButtonConfigurationMode.LegacyWithGear
 						CraftingGUI.forceSpecificRecipeResort = true;
 					} else if (MagicStorageConfig.RecipeBlacklistEnabled && Main.keyState.IsKeyDown(Keys.LeftControl)) {
 						if (recipeButtons.Choice == CraftingGUI.RecipeButtonsBlacklistChoice) {
-							if (storagePlayer.HiddenRecipes.Remove(obj.StoredItem)) {
-								Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealed", Lang.GetItemNameValue(obj.StoredItem.type)));
-
-								slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
-							}
+							Action<StoragePlayer, Item> revealRecipe = Main.keyState.IsKeyDown(Keys.LeftShift) ? RevealGlobalRecipe : RevealPlayerRecipe;
```

Comment:
Looking at it like this, I now think the below would be easier to understand, and also allows you to remove unused parameters
```cs
if (Main.keyState.IsKeyDown(Keys.LeftShift)) {
    RevealGlobalRecipe(item);
} else {
    RevealPlayerRecipe(storagePlayer, item);
}

Reply:
Sure, `Action<>` also comes with additional performance overhead.